onnx/defs/schema.cc

```diff
@@ -362,14 +362,103 @@ OpSchema& OpSchema::Attr(const Attribute& attr) {
 }
 
 OpSchema& OpSchema::Attr(
-    const char* name,
-    const char* description,
+    const std::string& name,
+    const std::string& description,
     AttributeProto::AttributeType type,
     bool required) {
   Attr(Attribute{name, description, type, required});
   return *this;
 }
 
+#define ATTR_SETTER_WITH_SINGLE_VALUE(type, field, attrtype)        \
```

Comment:
nit - might be more readable if you proxy to an overloaded template function to do the setting of an actual value and make the Attr() method templated on default-value (I think that's what Lu originally suggested)

Reply:
Don't quite get the point. Now the Attr APIs having type parameter there. like,
Attr(const std::string& name, const std::string& description, AttributeProto::AttributeType attr_type, const int64_t& default_value);

My original PR commit is Attr(const std::string& name, const std::string& description, const int64_t& default_value); because the attr_type in this case MUST be AttributeProto::INT. But Lu's right, adding the attr_type may be more readable although it's a dup information. That's why now I'm adding type check in-place to ensure that the attr_type and default_value's type are consistent.